src/model/Series.ts

```diff
@@ -582,26 +601,42 @@ class SeriesModel<Opt extends SeriesOption = SeriesOption> extends ComponentMode
     }
 
     private _innerSelect(data: SeriesData, innerDataIndices: number[]) {
-        const selectedMode = this.option.selectedMode;
+        const option = this.option;
+        const selectedMode = option.selectedMode;
         const len = innerDataIndices.length;
         if (!selectedMode || !len) {
             return;
         }
 
-        if (selectedMode === 'multiple') {
-            const selectedMap = this.option.selectedMap || (this.option.selectedMap = {});
+        if (selectedMode === 'series') {
+            option.selectedMap = 'all';
+        }
+        else if (selectedMode === 'multiple') {
+            if (!zrUtil.isObject(option.selectedMap)) {
+                option.selectedMap = {};
+            }
+            const selectedMap = option.selectedMap;
             for (let i = 0; i < len; i++) {
                 const dataIndex = innerDataIndices[i];
+                if (data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled'])) {
```

Comment:
I think we can remove the check here and only do the check in the `isSelected` method. The downside the stored `selectedMap` and `_selectedDataIndicesMap` will ignore the `select.disabled`. But it becomes consistent between `'series'`, `'multiple'` and `'single'` and it's simpler.